ocaml/xapi/xapi_pool_update.ml

```diff
@@ -182,17 +236,21 @@ let guidance_from_string = function
   | "restartXAPI" -> `restartXAPI
   | _ -> raise Bad_update_info
 
-let extract_update_info ~__context ~self ~verify ~destroy =
-  let host = Helpers.get_localhost ~__context in
+let extract_update_info ~__context ~vdi ~verify =
+  let vdi_uuid = Db.VDI.get_uuid ~__context ~self:vdi in
   finally
     (fun () ->
-      let update_path = attach ~__context ~self ~host in
+      let url = attach_helper ~__context ~uuid:vdi_uuid ~vdi in
       let update_path =
-        if String.startswith "file://" update_path then
-          String.sub_to_end update_path (String.length "file://")
+        if String.startswith "http://" url then
```

Comment:
This makes no sense. The URL has just been constructed above from the uuid you've passed in, but you're extracting it again here?